Cargo and secret cargo

LudvigStridsberg opened the pull request:
This pull request includes the cargo and secret cargo components as well as the fact that the trash-image files has been renamed.

LudvigStridsberg committed:
Extra css and elements

Created new css and elements for styling purposes.

LudvigStridsberg committed:
Element-types, constants and styling

Made the elements that will display the trash/tokens to img since we will use images and updated the element-references in cargo.js and secret-cargo.js. Also added more css-rules to make the elements look better.

LudvigStridsberg committed:
Renamed and centered trash-images

Renamed the trash-images and made the items be at the center of their canvas. Changed the CSS so that it looks better on several screens and changed how the source of the images are set in cargo.js and secret-cargo.js.

LudvigStridsberg committed:
Added timer-function

Added a function that counts down the time the player has left on their secret cargo.

LudvigStridsberg committed:
Added new Image and made background color match team

Added new Image and made background color match team.

LudvigStridsberg committed:
Added elements equired for timer

Added spans that can be used for displaying the time the user has left to use the secret cargo.

LudvigStridsberg committed:
Made strings single-quote

Replaced double quotation marks with single ones.

LudvigStridsberg committed:
Added visual infomation

Added visual information in the form of the opacity getting lower when the secret cargo isn't active by subscribing to the secret-cargo:faded event and switching classes when the event is recieved. Also playing around a bit with the timing of when the timer runs out/when the cargo is set as inactive.

LudvigStridsberg committed:
Fixed timing

Fixed the timing of the timer so that it runs out as expected. ALso fixed the width of the timer-element so that it doesn't distractingly changes the size of the secret cargo.

trumtomte commented on src/components/cargo.js:
```diff
@@ -19,7 +26,7 @@ function Cargo(el, context) {
     const token = tokens[cargo.token] ? tokens[cargo.token].name : '-';
 
-    el.textContent = `Cargo: ${token}`;
+    cargoSlot.src = `../../assets/${token}.png`;
```
Change all paths to assets to `/assets/...`, ie. an absolute path. Since this is handled by our HTTP-router, incase we would change the folder name or something else.

trumtomte replied:
Don't forget to test it also!

LudvigStridsberg replied:
Implemented and tested (it worked).

trumtomte reviewed (changes requested):
One minor thing to fix, otherwise it looks great! 👍🏻

LudvigStridsberg committed:
Fixed eslint errors

Fixed the errors that showed up in the files that I've worked in.

LudvigStridsberg committed:
Implemented requested changes

Changed the trash-image links in play.js, cargo.js and secret-cargo.js so that they use a absolute path.

trumtomte commented on src/components/cargo.js:
```diff
@@ -26,7 +26,7 @@ function Cargo(el, context) {
     const token = tokens[cargo.token] ? tokens[cargo.token].name : '-';
 
-    cargoSlot.src = `../../assets/${token}.png`;
+    cargoSlot.src = `assets/${token}.png`;
```
Nitpick: this is a relative path, remember absolute paths are either `/..../.../` or the with the protocol `http://....`.

trumtomte reviewed (changes requested):
Small nitpick.

LudvigStridsberg committed:
Absolute paths

Implemented absolute paths (for real this time).

Co-Authored-By: Sebastian Bengtegård <sebastianbengtegard@gmail.com>

Commit:
Merge branch 'main' into Cargo_and_SecretCargo
